src/SegmentedString.chpl

```diff
@@ -1429,11 +1438,8 @@ module SegmentedString {
     The SegFunction called by computeOnSegments for bytesToUintArr
   */
   inline proc stringBytesToUintArr(values, rng) throws {
-      var concat: uint;
-      for v in values[rng] {
-        concat = concat<<8 + v:uint;
-      }
-      return concat;
+      var localSlice = new lowLevelLocalizingSlice(values, rng);
+      return | reduce [i in 0..#rng.size] (localSlice.ptr(i):uint)<<(8*(rng.size-1-i));
```

Comment:
I know this should be equivalent to a more explicit forall, something like
```chapel
var concat: uint;
forall i in 0..#rng.size with (| reduce concat) {
  concat |= (localSlice.ptr(i):uint)<<(8*(rng.size-1-i));
}
return concat;
```
@ronawho @bradcray, I was wondering if there's any performance benefit between the two? I feel like the one line version could save on memory because (i don't think) we would need shadow variables?

Reply:
Also, I don't feel like the localizing slice should be necessary since we're calling using `computeOnSegments`, so things should be local? But I know a lot of the other computeOnSegments functions call either `interpretAsString` or `interpretAsBytes` which do a localizing slice, so it seemed worth a try

Reply:
@pierce314159 : Sorry for the delayed response, I was out most of last week.  I think the difference between your new code in 1441-1442, and your more explicit forall in the comment above should be a wash.  Specifically, Chapel typically rewrites `op reduce operand-expr` expressions into the equivalent `forall` loop with shadow variables, so as long as the expression driving the `forall` loop and the `operand-expr` are the same, you should feel free to use whichever style you prefer.  The shadow variables in cases like this are very modest in size (a scalar per task), so should not be a big deal.

I'm not terribly familiar with `lowLevelLocalizingSlice()`, so also found myself wondering whether this could just be written as something like:  `return | reduce [i in 0..<rng.size] (values[i]:uint)<<(8*(rng.size-1-i));` without much loss in performance by eliminating the need for the slice.  Or, if you're correct that all indices `i` are local, `values.localAccess(i)` instead of `values[i]`.  This paragraph is very much a shot in the dark, though, and if the performance is good enough and the style doesn't bother you, definitely ignorable.